saleor/static/dashboard-next/components/SaveFilterTabDialog/SaveFilterTabDialog.tsx

```diff
@@ -30,44 +30,58 @@ const SaveFilterTabDialog: React.FC<SaveFilterTabDialogProps> = ({
   onClose,
   onSubmit,
   open
-}) => (
-  <Dialog onClose={onClose} open={open} fullWidth maxWidth="sm">
-    <DialogTitle>
-      {i18n.t("Save Custom Search", {
-        context: "save filter tab"
-      })}
-    </DialogTitle>
-    <Form initial={initialForm} onSubmit={onSubmit}>
-      {({ change, data, submit }) => (
-        <>
-          <DialogContent>
-            <TextField
-              fullWidth
-              label={i18n.t("Search Name", {
-                context: "save search"
-              })}
-              name={"name" as keyof SaveFilterTabDialogFormData}
-              value={data.name}
-              onChange={change}
-            />
-          </DialogContent>
-          <DialogActions>
-            <Button onClick={onClose}>
-              {i18n.t("Cancel", { context: "button" })}
-            </Button>
-            <ConfirmButton
-              transitionState={confirmButtonState}
-              color="primary"
-              variant="contained"
-              onClick={submit}
-            >
-              {i18n.t("Save")}
-            </ConfirmButton>
-          </DialogActions>
-        </>
-      )}
-    </Form>
-  </Dialog>
-);
+}) => {
+  const [errors, setErrors] = React.useState(false);
+  const handleErrors = data => {
+    if (data.name.length) {
+      onSubmit(data);
+      setErrors(false);
+    } else {
+      setErrors(true);
+    }
+  };
```

Comment:
I get a feeling that we should do it in a more generic way, something like `useFormValidation` which would validate form data using [this lib](https://www.npmjs.com/package/yup), then return `UserError[]` with bound messages.

Reply:
@benekex2 Please open an issue for changing this in the future.